addons/mail/static/tests/message/message_tests.js

```diff
@@ -1126,6 +1126,73 @@ QUnit.test("allow attachment image download on message", async () => {
     await contains(".o-mail-AttachmentImage .fa-download");
 });
 
+QUnit.test("Can download all files of a message", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "test" });
+    const [attachmentId_1, attachmentId_2, attachmentId_3] = pyEnv["ir.attachment"].create([
+        { name: "test.png", mimetype: "image/png" },
+        { name: "Blah.png", mimetype: "image/png" },
+        { name: "shut.png", mimetype: "image/png" },
+    ]);
+    pyEnv["mail.message"].create([
+        {
+            attachment_ids: [attachmentId_1, attachmentId_2],
+            body: "<p>Test</p>",
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+        {
+            attachment_ids: [attachmentId_3],
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+    ]);
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("[title='Expand']:eq(0)");
+    await contains("[title='Download Files']");
+    await click("[title='Expand']:eq(1)");
+    await contains("[title='Download Files']", { count: 0 });
+});
+
+QUnit.test("Can remove files of message individually", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "test" });
+    const [attachmentId_1, attachmentId_2] = pyEnv["ir.attachment"].create([
+        { name: "attachment1.txt", mimetype: "text/plain" },
+        { name: "attachment2.txt", mimetype: "text/plain" },
+    ]);
+    pyEnv["mail.message"].create([
+        {
+            attachment_ids: [attachmentId_1, attachmentId_2],
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+        {
+            attachment_ids: [attachmentId_2],
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+        {
+            attachment_ids: [attachmentId_1],
+            body: "<p>Test</p>",
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+    ]);
```

Comment:
Note that currently it's not possible for 2 messages to share attachments.
Maybe we could support in the future... But just a small remark that test do not really represent a scenario that can be reproduced with current modelling of Discuss ;)